### PR TITLE
Clean up state tree to be just a HAMT of actor states.

### PR DESCRIPTION
**Pull request by anorth**
Replace incorrect implicit receiver actor creation with a placeholder and comment.
Fix accidental persistence of CallSeqNum and gas burn when implicit receiver creation fails.

FYI @icorderi

### CODE REVIEW

File: src/systems/filecoin_vm/interpreter/vm_interpreter.go [hunk 1]
```diff
@@ -174,32 +180,27 @@ func (vmi *VMInterpreter_I) ApplyMessage(
 		return
 	}
 
-	// make sure this is the right message order for fromActor
-	if message.CallSeqNum() != fromActor.CallSeqNum() {
-		_applyError(inTree, exitcode.InvalidCallSeqNum, SenderResolveSpec_Invalid)
+	// Check receiving actor and method exists, possibly creating an account actor.
+	// If this succeeds, compTreePreSend will become a state snapshot which includes
+	// implicit receiver creation, the sender (rather than miner) paying gas, and the sender's
+	// CallSeqNum being incremented; at least that much state change will be persisted even if the
```
Comment (sternhenri) [inline src/systems/filecoin_vm/interpreter/vm_interpreter.go]: What's the rationale for persisting that state?

Comment (anorth) [reply src/systems/filecoin_vm/interpreter/vm_interpreter.go]: Someone has to pay gas. This marks the transition from the miner being on the hook for a bad message, to the sender for a possibly-ok one.

File: src/systems/filecoin_vm/interpreter/vm_interpreter.go [hunk 2]
```diff
@@ -225,6 +226,29 @@ func (vmi *VMInterpreter_I) ApplyMessage(
...
+	}
+	if !message.To().IsKeyType() {
+		// Don't implicitly create an account actor for an address without an associated key.
+		return tree, false
+	}
+	if message.Method() != actor.MethodSend {
+		// Don't implicitly create account actor if message was expecting to invoke a method.
+		return tree, false
+	}
+	// TODO Create a new account actor via the InitActor, which will receive a new ID address
```
Comment (jzimmerman) [inline src/systems/filecoin_vm/interpreter/vm_interpreter.go]: nit: would be better to use `util.TODO()` or `panic` here, to make absolutely clear that this TODO is mandatory

File: src/systems/filecoin_vm/runtime/runtime.go [hunk 3]
```diff
@@ -429,7 +433,7 @@ func (rtOuter *VMContext) _sendInternal(input InvocInput, errSpec ErrorHandlingS
 		rtOuter._toplevelSender,
 		rtOuter._toplevelBlockWinner,
 		rtOuter._toplevelSenderCallSeqNum,
-		rtOuter._internalCallSeqNum+1,
+		rtOuter._internalCallSeqNum,
```
Comment (jzimmerman) [inline src/systems/filecoin_vm/runtime/runtime.go]: [Blocking] Why was the +1 removed? The internal CallSeqNum needs to be incremented somewhere, independently from the toplevel CallSeqNum.

File: src/systems/filecoin_vm/interpreter/vm_interpreter.go (hunk 2 above)
Comment (anorth) [reply src/systems/filecoin_vm/interpreter/vm_interpreter.go]: The implementation is coming in the direct follow-up, so I beg off this time.

File: src/systems/filecoin_vm/runtime/runtime.go (hunk 3 above)
Comment (anorth) [reply src/systems/filecoin_vm/runtime/runtime.go]: Sorry, I misread. Reverted.

**Pull request merged**